pkg/crypto/edg.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package crypto
+
+import "crypto/ecdsa"
+
+// Secp256k1EDG aggregates private key cryptography functions that employ secp256k1
+type Secp256k1EDG struct{}
```

Comment:
Since both Secp256k1EDG and Secp256r1EDG are static, they can be constructed as singletons for example similarly as base64 is constructing different encoding variants 
https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/encoding/base64/base64.go;l=99. This would remove the need to construct them every time when they are used. They do not occupy any memory (as they are struct{} type), but I think it would be more semantically correct.

Also, repeating tests TestEncodeSecp256r1PrivateKey TestGenerateSecp256r1Key and similars for Secp256k1EDG and Secp256r1EDG would ensure the correctness in time, however that seems trivial.